use pytest-split

jstriebel opened the pull request:
### Description:
Use [pytest-split](https://jerry-git.github.io/pytest-split) to distribute tests to multiple github action runners. This does not change local usage. Also updated the Contributing docs.


### Todos:
 - [x] Updated Documentation

jstriebel commented on .github/workflows/ci.yml:
```diff
@@ -156,18 +157,21 @@ jobs:
         poetry install --extras all
 
     - name: Check formatting
+      if: matrix.group == 1
       run: ./format.sh check
 
     - name: Lint code
+      if: matrix.group == 1
       run: ./lint.sh
 
     - name: Check typing
+      if: matrix.group == 1
       run: ./typecheck.sh
```
It might also make sense to have this in an extra job, but seems fine for now.

normanrz reviewed (approved):
Looks good. How is `.test_durations` updated (maybe add to readme or dev docs)?

jstriebel commented:
> How is `.test_durations` updated (maybe add to readme or dev docs)?

Good point, forgot to add the file, I updated the `CONTRIBUTING.md` (also [rendered in the docs](https://docs.webknossos.org/webknossos-py/CONTRIBUTING.html)), added now.
